pkg/kapp/resources/resources.go

```diff
@@ -169,6 +169,24 @@ func (c *ResourcesImpl) All(resTypes []ResourceType, opts AllOpts) ([]Resource,
 				}
 			}
 
+			// In case we successfully get the list for limited scope, we will
+			// try to run namespace scoped lists for rest of the allowed namespaced
+			if resType.Namespaced() && nsScopeLimited {
```

Comment:
On a second thought, is this not exactly what `c.allForNamespaces` does? Should we refactor the code above as well to deduce as much as we can?

Reply:
The case above seems to be when we do not have permission to list in all ns
And here it is because flag specifies this behaviour.

But the end result should be what `c.allForNamespaces` does in both cases I believe